model-engine/model_engine_server/inference/post_inference_hooks.py

```diff
@@ -76,6 +83,69 @@ def handle(
                 assert 200 <= res.status_code < 300
 
 
+class LoggingHook(PostInferenceHook):
+    def __init__(
+        self,
+        endpoint_name: str,
+        bundle_name: str,
+        user_id: str,
+        endpoint_id: Optional[str],
+        endpoint_type: Optional[ModelEndpointType],
+        bundle_id: Optional[str],
+        labels: Optional[Dict[str, str]],
+    ):
+        super().__init__(endpoint_name, bundle_name, user_id)
+        self._endpoint_id = endpoint_id
+        self._endpoint_type = endpoint_type
+        self._bundle_id = bundle_id
+        self._labels = labels
+
+    def handle(
+        self,
+        request_payload: EndpointPredictV1Request,
+        response: Dict[str, Any],
+        task_id: Optional[str],
+    ):
+        sts_client = boto3.client("sts", region_name=infra_config().default_region)
+        assumed_role_object = sts_client.assume_role(
+            RoleArn=infra_config().firehose_role_arn,
+            RoleSessionName="AssumeRoleSession1",
+        )
+        credentials = assumed_role_object["Credentials"]
+        session = boto3.Session(
+            aws_access_key_id=credentials["AccessKeyId"],
+            aws_secret_access_key=credentials["SecretAccessKey"],
+            aws_session_token=credentials["SessionToken"],
+        )
+        firehose_client = session.client("firehose", region_name=infra_config().default_region)
+        if (
+            not self._endpoint_id
+            or not self._endpoint_type
+            or not self._bundle_id
+            or not self._labels
+        ):
+            logger.warning(
+                "No endpoint_id, endpoint_type, bundle_id, or labels specified for request."
+            )
+            return
+        response["task_id"] = task_id
+        data_record = {
+            "REQUEST_BODY": request_payload.json(),
+            "RESPONSE_BODY": response,
+            "ENDPOINT_ID": self._endpoint_id,
+            "ENDPOINT_NAME": self._endpoint_name,
+            "ENDPOINT_TYPE": self._endpoint_type.value,
+            "BUNDLE_ID": self._bundle_id,
+            "LABELS": self._labels,
+        }
+        data = json.dumps(data_record)
+        firehose_response = firehose_client.put_record(
+            DeliveryStreamName=LOGGING_FIREHOSE_STREAM, Record={"Data": data.encode("utf-8")}
+        )
+        assert firehose_response["ResponseMetadata"]["HTTPStatusCode"] == 200
+        logger.info(f"Logged to firehose stream {LOGGING_FIREHOSE_STREAM}: {data_record}")
```

Comment:
saw that firehose_response should return a `RecordId` that could be usefull for logging